accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java

```diff
@@ -0,0 +1,38 @@
+package org.vertexium.accumulo.util;
+
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.client.impl.ClientContext;
+import org.apache.accumulo.core.client.impl.ConnectorImpl;
+import org.vertexium.VertexiumException;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+public class ConnectorUtils {
+    public static ClientContext getContext(Connector connector) {
+        if (!(connector instanceof ConnectorImpl)) {
+            throw new VertexiumException("Invalid connector type. Expected " + ConnectorImpl.class.getName() + " found " + connector.getClass().getName());
+        }
+        try {
+            Field f = ConnectorImpl.class.getDeclaredField("context");
+            f.setAccessible(true);
+            return (ClientContext) f.get(connector);
+        } catch (Exception e) {
+            throw new VertexiumException("Could not get context field", e);
+        }
+    }
+
+    public static String getTableId(Connector connector, String tableName) throws TableNotFoundException {
+        if (!(connector instanceof ConnectorImpl)) {
+            throw new VertexiumException("Invalid connector type. Expected " + ConnectorImpl.class.getName() + " found " + connector.getClass().getName());
+        }
+        try {
+            Method m = ConnectorImpl.class.getDeclaredMethod("getTableId", String.class);
```

Comment:
Looks like you might be able to use this helper instead: https://github.com/apache/accumulo/blob/rel/1.9.3/core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java#L68

Reply:
I saw that, it's takes an `instance` which we don't keep around. Looks like in Accumulo 2.0 we can get that too... https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/clientImpl/ConnectorImpl.java#L96 (context comes from `getAccumuloClient()`)

Reply:
I think you should be able to ask the `Connector` for its `Instance`: https://github.com/apache/accumulo/blob/rel/1.9.3/core/src/main/java/org/apache/accumulo/core/client/Connector.java#L216

Reply:
done